incomplete_cooperative/bounds.py

```diff
@@ -71,6 +69,31 @@ def compute_bounds_superadditive_cached(game: BoundableIncompleteGame) -> None:
         game.set_upper_bound(upper_bound, Coalition(coalition))
 
 
+def compute_bounds_superadditive_monotone_approx_cached(game: BoundableIncompleteGame, repetitions: int) -> None:
+    """Compute the bounds given a superadditive incomplete game."""
+    assert game.is_value_known(Coalition(0))
+    assert game.is_value_known(Coalition(2**game.number_of_players - 1))
+    all_coalitions, all_sorted, coal_structure = _get_sub_super_coalition_structure(game.number_of_players)
+    unknown_sorted = all_sorted[np.logical_not(game.are_values_known()[all_sorted])]
+    for _ in range(repetitions):
+        for coalition in unknown_sorted:
+            super_coalitions = all_coalitions[coal_structure[coalition] == 1]
+            complementary_coalitions = coalition ^ super_coalitions
+            lower_bound = np.max(game.get_lower_bounds()[super_coalitions] + game.get_lower_bounds()[complementary_coalitions])
+            game.set_lower_bound(lower_bound, Coalition(coalition))
+        for coalition in unknown_sorted:
+            super_coalitions = all_coalitions[np.logical_or(coal_structure[coalition] == 2, coal_structure[coalition] == 0)]
```

Comment:
I don't get this line either...

Reply:
See my response to your previous comment. Here we want non-sharp supercoalitions (I think that on line 80, only strict super coalitions should suffice?)